Paintroid/src/main/java/org/catrobat/paintroid/ui/tools/DefaultShapeToolOptionsView.kt

```diff
@@ -0,0 +1,244 @@
+                onOutlineWidthChanged(currentProgress)
+            }
+
+            override fun onStartTrackingTouch(seekBar: SeekBar) = Unit
+
+            override fun onStopTrackingTouch(seekBar: SeekBar) = Unit
+        })
+        outlineWidthEditText.addTextChangedListener(object : TextWatcher {
+            override fun beforeTextChanged(charSequence: CharSequence, i: Int, i1: Int, i2: Int) =
+                Unit
+
+            override fun onTextChanged(charSequence: CharSequence, i: Int, i1: Int, i2: Int) = Unit
+
+            override fun afterTextChanged(editable: Editable) {
+                val sizeText = outlineWidthEditText.text.toString()
+                val sizeTextInt: Int = try {
+                    sizeText.toInt()
+                } catch (exp: NumberFormatException) {
+                    MIN_STROKE_WIDTH
+                }
+                outlineWidthSeekBar.progress = sizeTextInt
+            }
+        })
+    }
+
+    private fun onShapeClicked(shape: DrawableShape) {
+        callback?.setToolType(shape)
+        setShapeActivated(shape)
+    }
+
+    private fun onDrawTypeClicked(drawType: DrawableStyle) {
+        callback?.setDrawType(drawType)
+        setDrawTypeActivated(drawType)
+    }
+
+    private fun onOutlineWidthChanged(outlineWidth: Int) {
+        callback?.setOutlineWidth(outlineWidth)
+    }
+
+    private fun resetShapeActivated() {
+        val buttons = arrayOf<View>(squareButton, circleButton, heartButton, starButton)
+        for (button in buttons) {
+            button.isSelected = false
+        }
+    }
+
+    private fun resetDrawTypeActivated() {
+        fillButton.isSelected = false
+        outlineButton.isSelected = false
+    }
+
+    private fun setShapeToolDialogTitle(title: Int) {
+        shapeToolDialogTitle.setText(title)
+    }
+
+    override fun setShapeActivated(shape: DrawableShape) {
+        resetShapeActivated()
+        when (shape) {
+            DrawableShape.RECTANGLE -> {
+                squareButton.isSelected = true
+                setShapeToolDialogTitle(R.string.shape_tool_dialog_rect_title)
+            }
+            DrawableShape.OVAL -> {
+                circleButton.isSelected = true
+                setShapeToolDialogTitle(R.string.shape_tool_dialog_ellipse_title)
+            }
+            DrawableShape.HEART -> {
+                heartButton.isSelected = true
+                setShapeToolDialogTitle(R.string.shape_tool_dialog_heart_title)
+            }
+            DrawableShape.STAR -> {
+                starButton.isSelected = true
+                setShapeToolDialogTitle(R.string.shape_tool_dialog_star_title)
+            }
+        }
+    }
+
+    private fun setupUI(
+        fillTitle: Int,
+        squareButtonResource: Int,
+        circleButtonResource: Int,
+        heartButtonResource: Int,
+        starButtonResource: Int,
+        outlineWidthSeekBarVisibility: Int,
```

Comment:
I feel like, its better to just have one Parameter for the visibilty since its always either Gone or Visible for all parameters.
Or what do you think :)?

Reply:
done :)